// Test protos for the Open Reaction Database.
syntax = "proto3";

package ord_test;

message Scalar {
    int32 int32_value = 1;
    int64 int64_value = 2;
    float float_value = 3;
    string string_value = 4;
    bytes bytes_value = 5;
}

message RepeatedScalar {
    repeated float value = 1;
}

message Enum {
    enum EnumValues {
        UNSPECIFIED = 0;
        FIRST = 1;
        SECOND = 2;
    }
    EnumValues value = 1;
}

message RepeatedEnum {
    enum EnumValues {
        UNSPECIFIED = 0;
        FIRST = 1;
        SECOND = 2;
    }
    repeated EnumValues value = 1;
}

message Nested {
    message Child {
        float value = 1;
    }
    Child child = 1;
}

message RepeatedNested {
    message Child {
        float value = 1;
    }
    repeated Child children = 1;
}

message Map {
    map<string, float> values = 1;
}
